### DISCUSSION TITLE
Docker image to build for specific platforms?

### DISCUSSION THREAD

**Discussion by imxade** [beeware/briefcase#1965] [Ideas]
It would be great to have something similar to [docker-pyinstaller](https://github.com/batonogov/docker-pyinstaller),
a set of Docker images to build distributables for specific platforms from any platform.

**Comment by freakboy3742** [comment]
Have you *tried* Briefcase? It *does* use Docker to build Linux images for any platform. [Pass in the `--target` option](https://briefcase.readthedocs.io/en/stable/reference/platforms/linux/system.html#target) when running a briefcase command, and it will build for that platform.

**Reply by imxade** [reply]
--target is only for linux right?
I want to build for other platforms too

**Reply by freakboy3742** [reply]
Well... yes, `--target` is only available for Linux builds. Docker can't be used to virtualize macOS or Windows.

It *can* be used on macOS (ie to build a Linux package while on macOS), although only to build binaries for the same architecture as the host machine (ie you can't build an x86_64 binary on an M1 laptop). 

In theory, it *could* be used on Windows, but there are known issues with Windows path handling.

The ability to use Linux to build macOS and Windows binaries is limited by the ability to virtualize those platforms, and the platform-specific tools required to build installers. For these use cases, we recommend using CI platforms (like GitHub actions).

**Reply by imxade** [reply]
Got it
Do we have a sample workflow?

**Reply by freakboy3742** [reply]
[This](https://github.com/rmartin16/briefcase-test-toga-app/blob/windows-code-signing/.github/workflows/package.yml) is an example of a workflow that will generate a signed Windows MSI file as an artefact of a Github Action. That example is slightly old, as it installs a `code-signing-windows` branch of Briefcase from github; that is no longer required, as that branch has been fully merged into the stable Briefcase release.

Formally documenting this is on our TODO list; see #400.
